### PR TITLE
fix(ecosystem): Track metrics for issue detail ticket creation

### PR DESCRIPTION
**Pull request by GabeVillalobos**
Adds metric gathering and logging to ticket creation on the issue details page. These may need to be recategorized in the future as a different form of issue creation, but for now I think it's fine to track this along with our issue alert rule metrics.

### CODE REVIEW

**Review by sentaur-athena — APPROVED** [review]

File: src/sentry/api/endpoints/group_integration_details.py [hunk 1]
```diff
@@ -266,12 +266,25 @@ def post(self, request: Request, group, integration_id) -> Response:
...
+            )
+
+            try:
+                data = installation.create_issue(request.data)
+            except IntegrationFormError as exc:
+                lifecycle.record_halt(exc)
+                return Response(exc.field_errors, status=400)
+            except IntegrationError as e:
+                lifecycle.record_failure(e)
+                return Response({"non_field_errors": [str(e)]}, status=400)
```
Comment (Christinarlong) [inline src/sentry/api/endpoints/group_integration_details.py:287]: Q: What are examples of `IntegrationErrors`, like is that a failure on our side ?

File: src/sentry/api/endpoints/group_integration_details.py [hunk 2]
```diff
@@ -266,12 +266,25 @@
...
```
[the remark this answers is not in the document]
Comment (GabeVillalobos) [reply src/sentry/api/endpoints/group_integration_details.py:287]: Already existed in the previous pr, this is what propagates field error info to our users.

**Review by Christinarlong — APPROVED** [review]
lgtm 👀

File: src/sentry/api/endpoints/group_integration_details.py (hunk 1 above)
Comment (GabeVillalobos) [reply src/sentry/api/endpoints/group_integration_details.py:287]: IntegrationErrors are any errors from a provider that our integrations themselves can't classify as a "field" error. 

We have some internal handling in the base `IntegrationInstallation` class that checks any `ApiInvalidRequestError` responses our integration methods raise: https://github.com/getsentry/sentry/blob/6209d6fbf55839bb7a2f93ef65decbf495a64974/src/sentry/integrations/base.py#L484

Which call into the implementing integration class's `error_fields_from_json` method: https://github.com/getsentry/sentry/blob/6209d6fbf55839bb7a2f93ef65decbf495a64974/src/sentry/integrations/jira/integration.py#L491

File: src/sentry/api/endpoints/group_integration_details.py [hunk 3]
```diff
@@ -266,12 +266,25 @@ def post(self, request: Request, group, integration_id) -> Response:
...
+        ).capture() as lifecycle:
+            lifecycle.add_extras(
+                {
+                    "provider": integration.provider,
+                    "integration_id": integration.id,
+                }
+            )
+
+            try:
+                data = installation.create_issue(request.data)
```
Comment (iamrajjoshi) [inline src/sentry/api/endpoints/group_integration_details.py:281]: might be worth tracking with a different slug b/c i have a feeling we have different levels of success based on where the ticket is created from.

also wondering if we should have a SLO around the endpoint that allows the user to search for a ticket in the 3p in the issue details page.

**Review by iamrajjoshi — APPROVED** [review]

File: src/sentry/api/endpoints/group_integration_details.py (hunk 3 above)
Comment (GabeVillalobos) [reply src/sentry/api/endpoints/group_integration_details.py:281]: Yeah I think you're right 🤔 I'll add that now.

**Pull request merged**